src/silx/io/utils.py

```diff
@@ -682,7 +682,7 @@ def open(filename):  # pylint:disable=redefined-builtin
             raise IOError("URL '%s' unsupported. Try to install h5pyd." % filename)
         path = uri.path
         endpoint = "%s://%s" % (uri.scheme, uri.netloc)
-        if path.startswith("/"):
+        if not uri.scheme.startswith("http") and path.startswith("/"):
```

Comment:
It would be better to check the `http(s)` scheme early, then this leading `/` removal would not be needed.
Also I encountered an issue while testing this PR locally because the HSDS server I have access to is served as `example.com/hsds/`.

I'll make a PR to your branch to fix both.